src/server/tiering/disk_storage.cc

```diff
@@ -4,16 +4,68 @@
 
 #include "server/tiering/disk_storage.h"
 
+#include "base/flags.h"
 #include "base/io_buf.h"
 #include "base/logging.h"
 #include "server/error.h"
 #include "server/tiering/common.h"
+#include "util/fibers/uring_proactor.h"
+
+using namespace ::dfly::tiering::literals;
+
+ABSL_FLAG(uint64_t, registered_buffer_size, 512_KB,
+          "Size of registered buffer for IoUring fixed read/writes");
 
 namespace dfly::tiering {
 
+using namespace ::util::fb2;
+
+namespace {
+
+UringBuf AllocateTmpBuf(size_t size) {
+  size = (size + kPageSize - 1) / kPageSize * kPageSize;
+  uint8_t* buf = new (std::align_val_t(kPageSize)) uint8_t[size];
+  return UringBuf{{buf, size}, std::nullopt};
+}
+
+void DestroyTmpBuf(UringBuf buf) {
+  DCHECK(!buf.buf_idx);
+  ::operator delete[](buf.bytes.data(), std::align_val_t(kPageSize));
+}
+
+UringBuf PrepareBuf(size_t size) {
+  DCHECK_EQ(ProactorBase::me()->GetKind(), ProactorBase::IOURING);
+  auto* up = static_cast<UringProactor*>(ProactorBase::me());
+
+  UringBuf buf;
+  if (auto borrowed = up->RequestBuffer(size); borrowed)
+    return *borrowed;
+  else
+    return AllocateTmpBuf(size);
```

Comment:
I do not mind whatever policy you implement but we should track this so we could learn from this

Reply:
ping

Reply:
AllocateTmpBuf has a vlog